CONTRIBUTING.md

```diff
@@ -142,3 +125,59 @@ We have some of this in the Reducer code, but generally discourage it.
 - https://github.com/avohq/reasonml-code-style-guide
 - https://cs.brown.edu/courses/cs017/content/docs/reasonml-style.pdf
 - https://github.com/ostera/reason-design-patterns/
+
+# TypeScript style
+
+**Prefer `const` over `let`, never use `var`**
+`var` is deprecated in JS. `let` should only be used for mutable variables.
+
+**Use functional style, avoid classes**
+We use classes for outer-facing APIs, but most of the codebase should use plain immutable objects with functions act on those objects.
+
+**Use immutable types when it doesn't hurt the performance**
+Wrap object types in [Readonly](https://www.typescriptlang.org/docs/handbook/utility-types.html#readonlytype) or mark individual fields as `readonly`.
+
+**Don't use namespaces**
+Use native ES modules instead, as [recommended by TypeScript documentation](https://www.typescriptlang.org/docs/handbook/namespaces-and-modules.html#using-modules).
+
+**Avoid `any` as much as possible**
+It's almost always possible to type things properly with modern Typescript.
+
+**Always use `===` instead of `==`**
+Loose equality is [crazy](https://dorey.github.io/JavaScript-Equality-Table/unified/).
+
+**Don't use too many external libraries**
+Heuristics for deciding whether pulling an external library is worth it:
```

Comment:
Maybe flag that external libraries are particularly bad for the Squiggle Lang, as that itself is really meant to be a library, and should thus be pretty small.

I feel like external libraries are much less bad for the playground, or other things that will be embedded less, like "Squiggle Hub" app equivalents.

Reply:
Right. Though the playground is currently bundled with everything else in components through webpack, so any dependency on playground increases the bundle size for those who want to embed SquiggleChart or SquiggleEditor through unpkg or some other CDN.